pkg/middlewares/redirect/redirect_scheme.go

```diff
@@ -64,8 +64,17 @@ func rawURLScheme(req *http.Request) string {
 		scheme = schemeHTTPS
 	}
 
-	if value := req.Header.Get(xForwardedProto); value != "" {
-		scheme = value
+	if xProto := req.Header.Get(xForwardedProto); xProto != "" {
+		// X-Forwarded-Proto header can be set to ws(s) when the client sends a websocket
+		// connection upgrade request whereas the request is made through the HTTP(S) protocol.
+		// As this middleware supports only HTTP(S) requests, we ignore the websocket
+		// protocols and converts them to the HTTP(S) protocol.
```

Comment:
```suggestion
		// When the initial request is a connection upgrade request, X-Forwarded-Proto
		// header might have been set by a previous hop to ws(s), even though the actual
		// protocol used so far is HTTP(s).
		// Given that we're in a middleware that is only used in the context of HTTP(s)
		// requests, the only possible valid schemes are one of "http" or "https", so we
		// convert back to them.		
```

Reply:
I explicitly ask to Maxence to write this comment with semantic splitting and not the 80 char limit

Reply:
```suggestion
		// When the initial request is a connection upgrade request,
		// X-Forwarded-Proto header might have been set by a previous hop to ws(s),
		// even though the actual protocol used so far is HTTP(s).
		// Given that we're in a middleware that is only used in the context of HTTP(s) requests,
		// the only possible valid schemes are one of "http" or "https", so we convert back to them.		
```